### ISSUE TITLE
npo ai

### ISSUE THREAD

**Issue by PlagueDuck** [PlagueDuck/FirstAI#1]
import torch
import torch.nn as nn
import torch.optim as optim

from flask import Flask, render_template, request

app = Flask(__name__)

# Список для хранения поданных заявок
applications = []

@app.route('/')
def index():
    return render_template('application_form.html')

@app.route('/submit', methods=['POST'])
def submit():
    if request.method == 'POST':
        name = request.form['name']
        email = request.form['email']
        message = request.form['message']
        
        # Создайте объект заявки и добавьте его в список
        application = {'name': name, 'email': email, 'message': message}
        applications.append(application)
        
        return render_template('confirmation.html', name=name)

if __name__ == '__main__':
    app.run(debug=True)

# Пример простой нейронной сети
class InflationPredictionNN(nn.Module):
    def __init__(self, input_size, hidden_size, output_size):
        super(InflationPredictionNN, self).__init__()
        self.hidden_size = hidden_size
        self.hidden = nn.Linear(input_size, hidden_size)
        self.relu = nn.ReLU()
        self.output = nn.Linear(hidden_size, output_size)

    def forward(self, x):
        hidden = self.relu(self.hidden(x))
        output = self.output(hidden)
        return output

# Пример данных (курс центробанка в течение нескольких лет)
# Используюся данные инфляции Центробанка с 2018 по 2022
years = torch.tensor([[i] for i in range(10)], dtype=torch.float32)  # Года: 0, 1, ..., 5
inflation_rate = torch.tensor([[4.27, 3.05, 4.91, 8.39, 11.94]], dtype=torch.float32)  # Пример инфляции

# Инициализация модели, потерь и оптимизатора
input_size = 1  # 1 входной признак: год
output_size = 1  # 1 выходной признак: прогноз инфляции
hidden_size = 128  # Размер скрытого слоя
learning_rate = 0.01
epochs = 10000

model = InflationPredictionNN(input_size, hidden_size, output_size)
criterion = nn.MSELoss()
optimizer = optim.Adam(model.parameters(), lr=learning_rate)

# Обучение модели
for epoch in range(epochs):
    # Forward pass
    outputs = model(years)
    loss = criterion(outputs, inflation_rate)

    # Backward pass and optimization
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()

    if (epoch+1) % 10 == 0:
        print(f'Epoch [{epoch+1}/{epochs}], Loss: {loss.item():.4f}')

# Пример прогнозирования для 5 лет
future_year = torch.tensor([[5]], dtype=torch.float32)
predicted_inflation = model(future_year)
print(f'Прогноз инфляции для 5 лет: {predicted_inflation.item():.4f}')
